sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -700,7 +700,13 @@ case class RegExpReplace(subject: Expression, regexp: Expression, rep: Expressio
       m.region(position, source.length)
       result.delete(0, result.length())
       while (m.find) {
-        m.appendReplacement(result, lastReplacement)
+        try {
+          m.appendReplacement(result, lastReplacement)
+        } catch {
+          case e: Exception =>
```

Comment:
can't you catch more specific exception here?

Reply:
I'm not 100% sure. Based on what I can see in the underlying Java file, I think the most restrictive choice we can make is `RuntimeException` which is just one level above `Exception`. But I still think `Exception` is the safest choice since there might be potential errors that don't fall in the `RuntimeExceptions` category.

Reply:
at least catch non-fatal exceptions like:
```suggestion
          case NonFatal(e) =>
```